scipy/stats/_continuous_distns.py

```diff
@@ -10664,12 +10664,32 @@ def find_kappa(data, loc):
             r = np.sum(np.cos(loc - data))/len(data)
             # See gh-18128 for more information.
 
+            # The function r[0](kappa) := I[1](kappa)/I[0](kappa) is monotonic
+            # increasing from r[0](0) = 0 to r[0](+inf) = 1.  The partial
+            # derivative of the log likelihood function with respect to kappa
+            # is monotonic decreasing in kappa.
             if r > 0:
                 def solve_for_kappa(kappa):
                     return sc.i1e(kappa)/sc.i0e(kappa) - r
 
+                # For backward compatibility, return MAX_KAPPA if the MLE of
+                # kappa would otherwise be larger than MAX_KAPPA.  Note that
+                # r[0](MAX_KAPPA) = 1.0 for double precision float.
+                MAX_KAPPA = 1e16
+
+                # The bounds of the root of r[0](kappa) = r are derived from
+                # the bounds of r[0](x) given in [1, Eq. 11]:
+                #
+                #   x/(1+sqrt(x*x+1)) <= r[0](x) <= x/sqrt(x*x+4)
+                #
+                # [1] Amos, D. E. (1973).  Computation of Modified Bessel
+                #     Functions and Their Ratios.  Mathematics of Computation,
+                #     28(125): 239-251.
+                with np.errstate(divide='ignore'):
```

Comment:
When does the error occur? The following snippet executes without a warning for me:

```python
import numpy as np

r = 0.999999999999999
2*r/np.sqrt(1-r*r), 2*r/(1-r*r)
``` 

Side note: one thing I read in some other PR here a while back was that writing $a^2-b^2=(a+b)(a-b)$ typically yields better precision with floating point, so we could do `(1+r)(1-r)` instead of `1-r*r`. Not sure how relevant that is here though, feel free to ignore.

Reply:
It sounds on the safe side to use `(1+r)(1-r)`, and I'll use that.

And you raised a very good point about "stress testing" the root finder. It does reveal the current bounds are too "tight" numerically. Half the lower bound and double the upper bound appears to pass my stress tests. I'll put the test code in a separate comment for easy reference.